### PR TITLE
Common controller/user profile

### LINKED ISSUE #5: CommonController - userProfile
The following API endpoints must be implemented in 'CommonController' class:

1. userProfile - "/userprofile/{userId}"

This endpoint is used to get the details of any user in the Quora Application. This endpoint can be accessed by any user in the application.

It should be a GET request
This endpoint must request the path variable 'userId' as a string for the corresponding user profile to be retrieved and access token of the signed in user as a string in authorization Request Header.
If the access token provided by the user does not exist in the database throw 'AuthorizationFailedException' with the message code - 'ATHR-001' and message - 'User has not signed in'.
If the user has signed out, throw "AuthorizationFailedException" with the message code -'ATHR-002' and message -'User is signed out.Sign in first to get user details' .
If the user with uuid whose profile is to be retrieved does not exist in the database, throw 'UserNotFoundException' with the message code -'USR-001' and message -'User with entered uuid does not exist'.
Else, return all the details of the user from the database in the JSON response with the corresponding HTTP status.

### PR DESCRIPTION
**Pull request by ahans30**
Added CommonController to enable endpoint UserProfile which fetches user profiles from database.

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  private UserDao userDao;
+
+  public UserEntity getUserProfile(final String userUuid, final String authorizationToken)
+      throws AuthorizationFailedException,
+      UserNotFoundException {
+
+    UserAuthTokenEntity userAuthTokenEntity = userDao.getUserAuthToken(authorizationToken);
+    UserEntity userEntity = userDao.getUser(userUuid);
+
+    if (userAuthTokenEntity.getLogoutAt() != null || userAuthTokenEntity.getExpiresAt()
```
Comment (simpleprahalad) [inline quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java]: Please maintain the order of exceptions as per the requirement.

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+
+@Entity
+@Table(name = "user_auth_tokens", schema = "qoura")
```
Comment (simpleprahalad) [inline quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: Schema name should be "quora". Typo error
Table name should be "user_auth". Please check

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+
+    @Column(name = "CREATED_BY")
+    @NotNull
+    private String createdBy;
+
+
+    @Column(name = "CREATED_AT")
+    @NotNull
+    private ZonedDateTime createdAt;
```
Comment (simpleprahalad) [inline quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: This is not as per requirement document. Please make the changes accordingly.
Additional fields like createdAt and version are not required.

**Review by simpleprahalad — CHANGES_REQUESTED** [review]
RestExceptionHandler needs to be added.

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java (hunk 1 above)
Comment (ahans30) [reply quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java]: Have re-worked the order of exception as per the requirement document in the latest commit. Please check.

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java (hunk 2 above)
Comment (ahans30) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: Issue fixed in the latest commit. Good observation, thanks!

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java (hunk 3 above)
Comment (ahans30) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: Went through the database document and fixed the issue in the latest commit. Thanks!

**Comment by ahans30** [comment]
Thanks for raising these issues. Have fixed the respective issues in the latest commit. Please check.

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java (hunk 1 above)
Comment (simpleprahalad) [reply quora-service/src/main/java/com/upgrad/quora/service/business/UserProfileBusinessService.java]: Ok.

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java (hunk 2 above)
Comment (simpleprahalad) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: :)

File: quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java (hunk 3 above)
Comment (simpleprahalad) [reply quora-service/src/main/java/com/upgrad/quora/service/entity/UserAuthTokenEntity.java]: Ok. Good Job !!!

**Review by simpleprahalad — APPROVED** [review]
Changes look good enough for the current commit.

**Pull request merged**